v2/apiserver/internal/core/jobs.go

```diff
@@ -254,6 +269,13 @@ func (j *jobsService) Create(
 		return errors.Wrapf(err, "error retrieving event %q from store", eventID)
 	}
 	if _, ok := event.Worker.Job(job.Name); ok {
+		// If we have a match and this is a retry, the job must be one pre-selected
+		// to be cached (has succeeded and does not use a shared workspace),
+		// therefore, return now to prevent rescheduling.
+		if event.Labels != nil || event.Labels[RetryLabelKey] != "" {
+			return nil
+		}
```

Comment:
Suppose the job that is requested to be created varies from the original job in some way... we should detect that and return an error.

There are ways that this could happen...

Suppose a worker does something non-deterministic-- like creates a random string-- and uses it as input to a job (command argument or env var perhaps).

When we detect such a difference, we cannot reasonably use the cached result.

Reply:
Ah, indeed... looking at this next.

Reply:
I agree. I think the tricky thing in that scenario, however, will be how we invalidate the inherited result once and only once.

We _shouldn't_ replace a job that wasn't an inherited one or else a single worker could get away with creating job X over and over.

I think that should be doable, because I think you already added some field to the job status that points back to the original event. If that's non-empty, I suppose the job was inherited and is eligible to be replaced.

This could create a strange scenario where someone looking at a new event that's a retry of an existing event could see jobs that initially appear to have been successful (inherited status) and _then_ they "restart" because the worker re-submitted the jobs with different input than the originals...

It gets me thinking that maybe we need to add something along the lines of a `INHERITED_SUCCESS` job phase. Terrible name. We'll have to come up with something better. But the point would be for users who have retried an event to be able to easily distinguish whether a job just ran or whether its result was inherited from a previous run. I think this is probably something to think about in a follow-up.

Reply:
Hmmm, right.  For this PR, should we just leave as returning the conflict error?